test/clj/schema/experimental/generators_test.clj

```diff
@@ -0,0 +1,51 @@
+(ns schema.experimental.generators-test
+  (:use clojure.test)
+  (:require
+   [clojure.test.check.properties :as properties]
+   [clojure.test.check.generators :as check-generators]
+   [clojure.test.check.clojure-test :as check-clojure-test]
+   [schema.core :as s]
+   [schema.experimental.generators :as generators]))
+
+
+(def OGInner
+  {(s/required-key "l") [s/Int]
+   s/Keyword s/Str})
+
+(def OGInner2
+  {:c OGInner
+   :d s/Str})
+
+(def OGSchema
+  {:a [s/Str]
+   :b OGInner2})
+
+(def FinalSchema
+  {:a (s/eq ["bob"])
+   :b {:c (s/conditional (fn [{:strs [l]}] (and (every? even? l) (seq l))) OGInner)
+       :d (s/eq "mary")}})
+
+(deftest sample-test
+  (let [res (generators/sample
+             20 OGSchema
+             {[s/Str] (generators/always ["bob"])
+              s/Int ((generators/fmap #(inc (* % 2))) check-generators/int)}
+             {[s/Int] (comp (generators/such-that seq)
+                            (generators/fmap (partial mapv inc)))
+              OGInner2 (generators/merged {:d "mary"})})]
+    (is (s/validate (s/conditional #(= (count %) 20) [FinalSchema]) res))))
```

Comment:
:sheep: might be clearer to count and  validate schema separately.

also, would be kinda cool to also check that (= (count (distinct res)) 20)